src/base/bittorrent/private/filterparserthread.cpp

```diff
@@ -100,54 +128,58 @@ int FilterParserThread::parseDATFilterFile(QString m_filePath, libt::ip_filter &
         return ruleCount;
     }
 
-    unsigned int nbLine = 0;
-    while (!file.atEnd() && !m_abort) {
-        ++nbLine;
-        QByteArray line = file.readLine();
-        // Ignoring empty lines
-        line = line.trimmed();
-        if (line.isEmpty()) continue;
-        // Ignoring commented lines
-        if (line.startsWith('#') || line.startsWith("//")) continue;
+    const QByteArray data = file.readAll();
+    int start = 0;
+    int endOfLine = -1;
+
+    while (true) {
+        start = endOfLine + 1;
+        if (start >= data.size())
+            break;
+        if (data[start] == '#' || (data[start] == '/' && data[start + 1] == '/'))
+            continue;
 
-        // Line should be split by commas
-        QList<QByteArray> partsList = line.split(',');
+        endOfLine = data.indexOf('\n', start);
+        if (endOfLine == -1) break;
+
+        QVector<int> delimIndices = indicesOfDelimiters(data, ',', start, endOfLine);
 
         // Check if there is at least one item (ip range)
-        if (partsList.isEmpty())
+        if (delimIndices.isEmpty())
             continue;
 
         // Check if there is an access value (apparently not mandatory)
-        if (partsList.size() > 1) {
+        if (delimIndices.size()) {
             // There is possibly one
-            const int nbAccess = partsList.at(1).trimmed().toInt();
+            const QByteArray accesscode(data.constData() + delimIndices[0] + 1, (delimIndices[1] - 1) - (delimIndices[0] + 1) + 1);
+            const int nbAccess = accesscode.toInt();
             // Ignoring this rule because access value is too high
             if (nbAccess > 127)
                 continue;
         }
 
         // IP Range should be split by a dash
-        QList<QByteArray> IPs = partsList.first().split('-');
-        if (IPs.size() != 2) {
-            Logger::instance()->addMessage(tr("IP filter line %1 is malformed. Line is: %2").arg(nbLine).arg(QString(line)), Log::CRITICAL);
+        QVector<int> delimIP = indicesOfDelimiters(data, '-', start, delimIndices[0] - 1);
+        if (delimIP.size() != 1) {
+            //Logger::instance()->addMessage(tr("IP filter line %1 is malformed. Line is: %2").arg(nbLine).arg(QString(line)), Log::CRITICAL);
             continue;
         }
 
         libt::address startAddr;
-        if (!parseIPAddress(IPs.at(0), startAddr)) {
-            Logger::instance()->addMessage(tr("IP filter line %1 is malformed. Start IP of the range is malformed: %2").arg(nbLine).arg(QString(IPs.at(0))), Log::CRITICAL);
+        if (!parseIPAddress(trim(data, start, delimIP[0] - 1), startAddr)) {
```

Comment:
The implementation of `parseIPAddress()` above does not need trimmed from end string to parse the address. Anyway, since the buffer is writable, you can just put zeroes in place of delimiters instead of copying the string.